src/js/core/league.js

```diff
@@ -335,6 +335,16 @@ async function create(name, tid, leagueFile = {}, startingSeason, randomizeRoste
             }
 
             players.forEach(async p => {
+                if ("name" in p && !("firstName" in p) && !("lastName" in p)) {
```

Comment:
So the idea is to keep name, firstName, and lastName all as properties on the player object? That's good for backwards compatibility, but I'm not sure if I like it... maybe I do... need to think about it. Either way, this code should be in that augmentPartialPlayer function called right below.